src/main/java/org/commonjava/storage/pathmapped/datastax/CassandraPathDB.java

```diff
@@ -206,13 +227,31 @@ public void insert( PathMap pathMap )
 
         String path = normalize( parent, pathMap.getFilename() );
         PathMap prev = getPathMap( fileSystem, path );
-        if ( prev != null )
+        final boolean existedPathMap = prev != null;
+        if ( existedPathMap )
         {
             delete( fileSystem, path );
         }
 
+        final FileChecksum existedFileChecksum = getFileChecksumByChecksum( checksum );
+
+        if ( existedFileChecksum != null )
+        {
+            logger.info( "File checksum conflict, should use existed file storage" );
+            final PathMap existed = getPathMapByFileId( existedFileChecksum.getFileId() );
```

Comment:
I can say there are two issues. The mapping to a fileId is many-to-one, i.e., there might be multiple entries in pathMap table that map to one fileId. The other issue is "previousStorage = existed.getFileStorage()". It is set in following "setFileStorage" but reclaimed at the end. I guess it should be "reclaim(deprecatedFileId, newStorage);"?
Anyway, this is easy to correct. The main question is how to get the existing file storage. We can add a field "storage" to Checksum table. So each time we get by checksum, we get both the fileId and storage. This also eliminate the above ALLOW FILTERING problem.

Reply:
For first one, seems it is a typo..... I'll fix it.
For second one, seems it is good idea to add a storage field to avoid filtering. I'll add it here.